tests/unit/fork_test.cpp

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <unistd.h>
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/process.h>
+   #include "mocking.h"
+   int vfs_dup(fs_handle h, fs_handle *dup_h);
+   void vfs_close(fs_handle h);
+   int fork_dup_all_handles(struct process *pi);
+}
+
+class vfs_mock : public KernelSingleton {
+public:
+
+   MOCK_METHOD(int, vfs_dup, (fs_handle h, fs_handle *dup_h), (override));
+   MOCK_METHOD(void, vfs_close, (fs_handle h), (override));
+};
+
+TEST(fork_dup_all_handles, trigger_inside_path) {
+   vfs_mock mock;
+   process pi;
+   bzero(&pi, sizeof(process));
+   fs_handle_base handle0_dup, handle1_dup, handle2_dup;
+
+   pi.handles[0] = (fs_handle) &handle0_dup;
+   pi.handles[1] = (fs_handle) &handle1_dup;
+   pi.handles[2] = (fs_handle) &handle2_dup;
+
+   EXPECT_CALL(mock, vfs_dup(_, _))
+      .WillOnce(
+         DoAll(
+               SetArgPointee<1>(&handle0_dup),
+               Return(0)
+         )
+      )
+      .WillOnce(
+         DoAll(
+               SetArgPointee<1>(&handle1_dup),
+               Return(0)
+         )
+      )
+      .WillOnce(Return(-1));
+
+   EXPECT_CALL(mock, vfs_close(&handle0_dup));
```

Comment:
Theoretically, we could totally fake the dup handles and make `vfs_dup()` to return 0xbb00, 0xbb01, 0xbb02. Not sure what's best, really. Maybe allocate 3 actual handles + 2 dup handles, just to stay closer to reality, even if those pointers are never de-referenced.